torch/utils/data/dataset.py

```diff
@@ -152,6 +163,24 @@ def __add__(self, other: Dataset[T_co]):
     # No `def __len__(self)` default?
     # See NOTE [ Lack of Default `__len__` in Python Abstract Base Classes ]
 
+    def __getattr__(self, attribute_name):
+        if attribute_name in IterableDataset.functions:
+            function = functools.partial(IterableDataset.functions[attribute_name], self)
+            return function
+        else:
+            raise AttributeError
```

Comment:
We can actually extend it to some factory functions without `self` argument, such as the source DataPipes, maybe by add another registration to `factory_functions`
```py
torch.utils.data.IterDataPipe.from_Hive('path').filter(filter_fn = lambda x: x % 2 == 1).map(fn = lambda x: x * 10)
```
DataPipe can registered:
```py
@register_factory_datapipe('from_Hive')
class HiveIterDataPipe(IterDataPipe):
```

Reply:
Let me think about it. Anyway we can pass on it now and add later on.